src/eduid/webapp/idp/views/next.py

```diff
@@ -276,6 +282,58 @@ def _set_user_options(res: AuthnOptions, eppn: str) -> None:
     return None
 
 
+def _geo_statistics(ticket: LoginContext, sso_session: Optional[SSOSession]) -> None:
+    """ Log user statistics from login event """
+
+    if not sso_session:
+        return None
+
+    if not current_app.conf.geo_statistics_secret_key or not current_app.conf.geo_statistics_url:
+        return None
+
+    ua = get_user_agent()
+    if not ua:
+        return None
+
+    if ua.parsed.browser.family in ['Python Requests', 'PingdomBot'] or ua.parsed.is_bot:
+        return None
+
+    secret = bytes(current_app.conf.geo_statistics_secret_key, 'utf-8')
+
+    user_hash = hmac.HMAC(secret, hashes.SHA256())
+    user_hash.update(bytes(sso_session.eppn, 'utf-8'))
+
+    device_id = None
+    if ticket.known_device:
+        device_id_hash = hmac.HMAC(secret, hashes.SHA256())
+        device_id_hash.update(bytes(ticket.known_device.state_id, 'uft-8'))
+        device_id = device_id_hash.finalize().hex()
+
+    d: Dict[str, Any] = {
+        'data': {
+            'user_id': user_hash.finalize().hex(),
+            'client_ip': request.headers.get('x-forwarded-for'),
+            'device_id': device_id,
+            'user_agent': {'browser': {}, 'os': {}, 'device': {}, 'sophisticated': {}},
+        }
+    }
+
+    data = d['data']
+    data['user_agent']['browser']['family'] = ua.parsed.browser.family
+    data['user_agent']['os']['family'] = ua.parsed.os.family
+    data['user_agent']['device']['family'] = ua.parsed.device.family
+    data['user_agent']['sophisticated']['is_mobile'] = ua.parsed.is_mobile
+    data['user_agent']['sophisticated']['is_pc'] = ua.parsed.is_pc
+    data['user_agent']['sophisticated']['is_tablet'] = ua.parsed.is_tablet
+    data['user_agent']['sophisticated']['is_touch_capable'] = ua.parsed.is_touch_capable
+
```

Comment:
Why do we want to have User-Agent data in the geo-based risk assessment research data? Maybe it's just the use of 'geo' that is a bit outdated?

Reply:
That's right, the name is outdated since the start, but since we know what it is maybe we can stick to it for the moment - until we have decided on a better name?